frontend/controller/service-worker.js

```diff
@@ -13,23 +15,95 @@ sbp('sbp/selectors/register', {
     // TODO: move ahead with encryption stuff ignoring this service worker stuff for now
     // TODO: improve updating the sw: https://stackoverflow.com/a/49748437
     // NOTE: user should still be able to use app even if all the SW stuff fails
+    if (!('serviceWorker' in navigator)) { return }
```

Comment:
Wouldn't it be prudent to check for everything needed here? For example, `typeof Notification !== 'undefined'`

Reply:
Agreed.